### DISCUSSION TITLE
go sdk的一些疑问

### DISCUSSION THREAD

**Discussion by jilieryuyi** [milvus-io/milvus#7479] [Q&A and General discussion] [answered]
是否需要每次insert后都调用Flush和LoadCollection，具体调用后发生了什么事情？没调用会如何？

求解！

**Comment by xiaofan-luan** [comment]
不需要每次都调用flush和load collection。正常的使用方式是插入数据，load，然后查询，后面的insert不需要load，由于milvus还缺少一些机制（我们正在做了），建议大量插入后可以release然后重新load

**Reply by jilieryuyi** [reply]
> 不需要每次都调用flush和load collection。正常的使用方式是插入数据，load，然后查询，后面的insert不需要load，由于milvus还缺少一些机制（我们正在做了），建议大量插入后可以release然后重新load

![1](https://user-images.githubusercontent.com/2398119/132090995-26960532-2f82-4579-b917-8f619ea77d60.png)
不执行 flush，这里的数量都是0 ~， 看不到实际的向量数量

**Comment by xiaofan-luan** [comment, marked as answer]
you need to write more data and it is automatically flushed, the stats is only updated when flush, but data is visible
